packages/dvilela/skills/memeooorr_abci/behaviour_classes/twitter.py

```diff
@@ -417,145 +496,262 @@ def get_event(self) -> Generator[None, None, str]:
 
-        tweets = yield from self.get_previous_tweets()  # type: ignore
-        tweets = tweets[-5:] if tweets else None  # type: ignore
+            mech_responses = self.synchronized_data.mech_responses
+            if not mech_responses:
+                self.context.logger.error("No mech responses found")
 
-        if tweets:
-            random.shuffle(tweets)
-            previous_tweets = "\n\n".join(
+            prompt = TWITTER_DECISION_PROMPT_WITH_MECH_RESPONSE.format(
+                persona=persona,
+                previous_tweets=previous_tweets,
+                other_tweets=other_tweets,
+                mech_response=mech_responses,
+                time=self.get_sync_time_str(),
+            )
+
+            llm_response = yield from self._call_genai(
+                prompt=prompt,
+                schema=build_decision_schema(),
+            )
+
+            # clearing the previous tweets and other tweets from db
+            self.context.logger.info(
+                "clearing all the pending tweets , interacted tweet id ,previous tweets and other tweets from db"
+            )
+            yield from self._write_kv({"previous_tweets_for_tw_mech": ""})
+            yield from self._write_kv({"other_tweets_for_tw_mech": ""})
+            yield from self._write_kv({"interacted_tweet_ids_for_tw_mech": ""})
+            yield from self._write_kv({"pending_tweets_for_tw_mech": ""})
+
+        else:
+            self.context.logger.info(
+                "No Mech for twitter detected , using prompt for decision and introducing tools to LLM"
+            )
+
+            other_tweets = "\n\n".join(  # type: ignore
                 [
-                    f"tweet_id: {tweet['tweet_id']}\ntweet_text: {tweet['text']}\ntime: {datetime.fromtimestamp(tweet['timestamp']).strftime('%Y-%m-%d %H:%M:%S')}"
-                    for tweet in tweets
+                    f"tweet_id: {t_id}\ntweet_text: {t_data['text']}\nuser_id: {t_data['user_id']}"
+                    for t_id, t_data in dict(
+                        random.sample(list(pending_tweets.items()), len(pending_tweets))
+                    ).items()
                 ]
             )
-        else:
-            previous_tweets = "No previous tweets"
 
-        prompt = TWITTER_DECISION_PROMPT.format(
-            persona=persona,
-            previous_tweets=previous_tweets,
-            other_tweets=other_tweets,
-            time=self.get_sync_time_str(),
-        )
+            tweets = yield from self.get_previous_tweets()  # type: ignore
+            tweets = tweets[-5:] if tweets else None  # type: ignore
+
+            if tweets:
+                random.shuffle(tweets)
+                previous_tweets = "\n\n".join(  # type: ignore
+                    [
+                        f"tweet_id: {tweet['tweet_id']}\ntweet_text: {tweet['text']}\ntime: {datetime.fromtimestamp(tweet['timestamp']).strftime('%Y-%m-%d %H:%M:%S')}"
+                        for tweet in tweets
```

Comment:
In general, this file needs a lot of cleanup. It just looks too dirty. We need encapsulation, good code comments, less nesting. I see a lot of type ignores. Be careful with always ignoring lint errors, we have them for a reason. Always ensure that you only ignore if it makes sense.